src/utils/redisCache.js

```diff
@@ -0,0 +1,79 @@
+let crypto = require('crypto')
+let _ = require('lodash/fp')
+let F = require('futil')
+let deterministic_stringify = require('json-stable-stringify')
+
+let redisCache = (config, schema) => {
+  let { redisCache, redisPrefix = 'es-cache:' } = config
+  let redisClient = F.maybeCall(config.getRedisClient)
+  let { caching: { ttlSecs, ttrSecs } = {} } = schema
+
+  // to resolve duplicate requests with the same promise
+  let instaCache = Object.create(null)
```

Comment:
Why not use a new Map()? They are much faster than objects =] since this is a speed thing.

Reply:
Thank you for the feedback!